crates/bevy_ecs/src/query/iter.rs

```diff
@@ -61,6 +201,38 @@ impl<'w, 's, Q: WorldQuery, F: ReadOnlyWorldQuery> Iterator for QueryIter<'w, 's
         let min_size = if archetype_query { max_size } else { 0 };
         (min_size, Some(max_size))
     }
+
+    #[inline]
+    fn fold<B, Func>(mut self, init: B, mut func: Func) -> B
+    where
+        Func: FnMut(B, Self::Item) -> B,
+    {
+        let mut accum = init;
+        // Empty any remaining uniterated values from the current table/archetype
+        while self.cursor.current_row != self.cursor.current_len {
+            let Some(item) = self.next() else { break };
+            accum = func(accum, item);
+        }
+        if Q::IS_DENSE && F::IS_DENSE {
+            for table_id in self.cursor.table_id_iter.clone() {
+                // SAFETY: Matched table IDs are guarenteed to still exist.
```

Comment:
```suggestion
                // SAFETY: Matched table IDs are guaranteed to still exist.
```

Reply:
I can never seem to spell this correctly.